src/com/owncloud/android/services/observer/InstantUploadsObserverFactory.java

```diff
@@ -0,0 +1,46 @@
+/**
+ *   ownCloud Android client application
+ *
+ *   @author David A. Velasco
+ *   Copyright (C) 2017 ownCloud GmbH.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License version 2,
+ *   as published by the Free Software Foundation.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.owncloud.android.services.observer;
+
+import android.content.Context;
+import android.os.Build;
+
+import com.owncloud.android.db.PreferenceManager.InstantUploadsConfiguration;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+/**
+ * Builds new instances of {@link InstantUploadsObserver}, using the appropritate implementation.
+ */
+class InstantUploadsObserverFactory {
+
+    private static final String TAG = InstantUploadsObserverFactory.class.getName();
+
+    public static InstantUploadsObserver newObserver(InstantUploadsConfiguration config, Context context) {
+        if (Build.MODEL.toLowerCase().contains("nexus") ||
```

Comment:
It is a "fast-&-dirty" check. But maybe it's good enough for production. Or maybe we go in the end with the method based on commons.io for all the devices. In the end, the idea that "current implementation works in Google devices but not in others" is not exact, it is a rough estimation based on the feedback we have from users and their devices. We do not know the real reason why ```FileObserver``` works in some devices and not in others. 

Moreover, for this concrete problem - the fail of ```FileObserver``` to observer, there is nothing that points to camera apps as responsible of the problem. 

Remember: at the moment we have two methods of detection of new pictures working at the same time in the app: ```FileObserver``` on the camera folder, and listening for ```NEW_PICTURE``` broadcasts. The second one is the one we know is affected by camera apps, not the first one.